src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -5,14 +5,104 @@ package freechips.rocketchip.diplomacy
 import Chisel.{defaultCompileOptions => _, _}
 import chisel3.internal.sourceinfo.{SourceInfo, UnlocatableSourceInfo}
 import chisel3.{MultiIOModule, RawModule, Reset, withClockAndReset}
-import chisel3.experimental.{ChiselAnnotation}
+import chisel3.experimental.{BaseModule, ChiselAnnotation}
+import chisel3.aop.{Select => CSelect}
 import firrtl.passes.InlineAnnotation
 import freechips.rocketchip.config.Parameters
 import freechips.rocketchip.util.CompileOptions.NotStrictInferReset
 
 import scala.collection.immutable.{ListMap, SortedMap}
 import scala.util.matching._
 
+object Select {
+  type AnyMixedNode = MixedNode[_, _, _, _ <: Data, _, _, _, _ <: Data]
+
+  case class InwardEdge[Bundle <: Data, EdgeInParams](
+    params: Parameters,
+    bundle: Bundle,
+    edge: EdgeInParams,
+    node: OutwardNode[_, _, Bundle],
+  )
```

Comment:
I see a lot of wildcard type parameters here. Is it possible someone may someday want to care about the first two type parameters of `node`? If so, we should probably go ahead and add them because adding them later will be very painful for anyone using this type.

That being said I don't know much about best practices and maintaining backwards compatibility in using type parameters.

Reply:
In this case, those parameters are kind of implicitly captured by the `EdgeInParams` type member, which inevitably will have members of the "missing" `DI` and `UI` types. So I guess I don't see much value in offering those through this API. I do wonder whether since these classes are part of the public `collect` interface, should they become part of top-level diplomacy API instead of embedded in this `object Select`?